### PR TITLE
Create user and user manager

### PR DESCRIPTION
**Pull request by lolivera-eb**
create the model user and managerUser, and integrate it with the login and logout 

settigs : AUTH_USER_MODEL 
models:: -User
                -UserManager

### CODE REVIEW

**Review by ecrespillo-eb — APPROVED** [review]

File: auth_app/models.py [hunk 1]
```diff
@@ -1 +1,46 @@
...
+
+
+class UserManager(BaseUserManager):
+    def create_user(self, email, username, password):
+        if not email:
+            raise ValueError("ENTER AN EMAIL BUDDY")
+        if not username:
+            raise ValueError("I KNOW YOU HAVE A NAME")
+        if not password:
+            raise ValueError("PASSWORD?!?!?!? HELLO??")
```
Comment (julianandres-eb) [inline auth_app/models.py:16]: Is this necessary? Django raises a `IntegrityError` when trying to store a `NULL` value in the DB

File: auth_app/models.py [hunk 2]
```diff
@@ -1 +1,46 @@
...
+
+        user = self.model(
+            email=self.normalize_email(email),
```
Comment (julianandres-eb) [inline auth_app/models.py:19]: Have you consider [this](https://docs.djangoproject.com/en/3.2/topics/auth/customizing/#django.contrib.auth.models.AbstractBaseUser.clean)?

File: auth_app/models.py [hunk 3]
```diff
@@ -1 +1,46 @@
...
+        user.is_superuser = True
+        user.save()
+        return user
+
+
+class User(AbstractBaseUser, PermissionsMixin):
+    email = models.EmailField(unique=True)
+    username = models.CharField(max_length=25, unique=True)
+    token = models.CharField(max_length=200)
+    # avatar = models.ImageField(blank=True, null=True)
```
Comment (julianandres-eb) [inline auth_app/models.py:37]: NIT: Remove dead code

**Comment by julianandres-eb** [comment]
I don't see any unit tests. Even if it's only Django models, it's better to have it for all type of PRs

**Pull request merged**